src/interfaces/ContinuousGenerator.ts

```diff
@@ -0,0 +1,179 @@
+import { Random } from "@reside-ic/random";
+import { Packer } from "../Packer.ts";
+import { ZeroEvery } from "../zero.ts";
+import { DelayResult, Delays } from "../delays.ts";
+
+/**
+ * Interface defining the functionality of a continuous time model, which can be used by {@link ContinuousSystem}
+ * to initialise and update particles. Generators are stateless and are always provided with all state and other
+ * parameters required to update the particle in each method.
+ *
+ * @typeParam TShared Values which are shared between all particles in a group and are not mutated by them -
+ * the model parameter values for that group
+ *
+ * @typeParam TInternal Internal state values which can be mutated by generators, used to improve efficiency of the
+ * system by e.g. caching calculation results for use by other particles.
+ */
+export interface ContinuousGenerator<TShared, TInternal> {
+    /**
+     * Sets the initial state of a particle.
+     *
+     * @param time The current time at which to initialise particle state
+     * @param shared The shared parameter values used by the particle's group
+     * @param internal The internal state used by the particle's group
+     * @param stateNext The array of values which should be updated by the generator with initial particle state values
+     * @param random A random number generator which may be used by the generator to initialise values
+     */
+    initial(time: number, shared: TShared, internal: TInternal, stateNext: number[], random: Random): void;
```

Comment:
A lot of these methods are shared with DiscreteGenerator - could put them in a base interface type..?